control/metrics.go

```diff
@@ -706,3 +563,123 @@ func addStandardAndWorkflowTags(m core.Metric, allTags map[string]map[string]str
 	}
 	return metric
 }
+
+// isTuple returns true when incoming namespace's element has been recognized as a tuple, otherwise returns false
+// notice, that the tuple is a string which starts with `core.TuplePrefix`, ends with `core.TupleSuffix` and contains at least one `core.TupleSeparator`
+// e.g. (host0;host1)
+func isTuple(element string) bool {
+	if strings.HasPrefix(element, core.TuplePrefix) && strings.HasSuffix(element, core.TupleSuffix) && strings.Contains(element, core.TupleSeparator) {
+		return true
+	}
+	return false
+}
+
+// containsTuple checks if a given element of namespace has a tuple; if yes, returns true and recognized tuple's items
+func containsTuple(nsElement string) (bool, []string) {
+	tupleItems := []string{}
+	if isTuple(nsElement) {
+		if strings.ContainsAny(nsElement, "*") {
+			// an asterisk covers all tuples cases (eg. /intel/mock/(host0;host1;*)/baz)
+			// so to avoid retrieving the same metric more than once, return only '*' as a tuple's items
+			tupleItems = []string{"*"}
+		} else {
+			tuple := strings.TrimSuffix(strings.TrimPrefix(nsElement, core.TuplePrefix), core.TupleSuffix)
+			tuple = strings.Replace(tuple, core.TupleSeparator, " ", -1)
```

Comment:
The reason why I replace `core.TupleSeparator` with space is allowing to use "strings.Fields" which removing white spaces, so all of the following cases are possible:
- no spaces: `/intel/mock/(host0;host1)/baz`
- spaces before and after semicolon: `/intel/mock/(host0 ; host1)/baz`
- spaces next to brackets: `/intel/mock/( host0 ; host1 )/baz`

Now I see that will not work for tuple's item consisting of two words. So, there is no sense to use "strings.Fields". Removing white spaces by using "string.TrimSpace" for each of tuple's items will give the same result and do not impact on two-word-item.

#### Conclusion:
This fragment:
```
tuple := strings.TrimSuffix(strings.TrimPrefix(nsElement, core.TuplePrefix), core.TupleSuffix)
tuple = strings.Replace(tuple, core.TupleSeparator, " ", -1)
tupleItems = strings.Fields(tuple)
```
will be changed to:
```
items := strings.Split(tuple, core.TupleSeparator)
 // removing all leading and trailing white space
 for _, item := range items {
  tupleItems = append(tupleItems, strings.TrimSpace(item))
}
```